File: src/app/models/post.ts
export interface Post {
  id: number;
  title: string;
  slug: string;
  // banner?: string;
  // filename?: string;
  summary: string;
  content: string;
  // tags: string[];
  featured: boolean;
  published: boolean;
  publishedAt: Date;
  createdAt: Date;
  updatedAt: Date;
}

File: app/models/post.rb
class Post < ApplicationRecord
  belongs_to :user
  acts_as_taggable_on :tags
  validates :title, :slug, presence: true

  scope :published,   -> { where(published: true) }
  scope :unpublished, -> { where(published: false) }

  def to_param
    slug
  end

  def previous_article
    self.class.published.where('id < ?', self.id).last
  end

  def next_article
    self.class.published.where('id > ?', self.id).first
  end
end
